HousingFinanceInterimApi/V1/UseCase/GenerateRentPositionUseCase.cs

```diff
@@ -120,5 +148,19 @@ private async Task<List<GoogleFileSettingDomain>> GetGoogleFileSetting(string la
 
             return googleFileSettings;
         }
+
+        /// <summary>
+        /// Filters out each file that is the last file for a financial year (31st March)
+        /// </summary>
+        static IEnumerable<File> getLastFilesForFinancialYears(IEnumerable<File> fileList)
+        {
+            var filesOnLastDayOfFinancialYear = fileList.Where(
+                f => f.CreatedTime != null
+                     && f.CreatedTime.Value.Day == 31
+                     && f.CreatedTime.Value.Month == 3
```

Comment:
Won't some of the years Have March 31st as Sunday? Like 2024 March 31st?
I doubt Hackney staff will be creating a file during Sunday. So chances are your file will be dated to 2024 March 29th

Reply:
Yes, Liudvikas is correct. The last day of the Financial year is the last weekday in March.

Reply:
The production Rent Position folder has files on weekends too though, isn't it an automated process?
https://drive.google.com/drive/u/0/folders/1gYJh48lLyTdkSPdK-fGRjYHhXySm78ae

Reply:
Could just be the source Google Sheet / CSV doesn't get edited on weekends, but it's still there?

Reply:
Ah I see - the end of financial year for processing purposes is the last working day, will adapt the function to reflect this.

Reply:
This has now been reflected in the code